### ISSUE TITLE
requires not evaluated before dynamic variables

### ISSUE THREAD

**Issue by hans-d** [go-task/task#1970]
### Description

Have a task that defines a `requires: { vars: [] }', and additionallyuse a dynamic variable.

The dynamic variables is resolved before the requires are checked.

### Version

Task version: v3.40.1 (h1:Q/TjPh5SAI+XK2fBravDPZXNGm6mvnwrLJqCXfhZ3Wc=)

### Operating system

Linux

### Example Taskfile

```YAML
version: '3'

tasks:
  test-ok:
    desc: Fails correctly
    requires:
      vars: [ FOO ]
    vars:
      BAR: '{{.FOO}}'
    cmds:
      - echo Got {{.FOO}} {{.BAR }}

  test-fail:
    desc: Should fail, but already executes sh `Command "echo '' && false" failed: exit status 1`
    requires:
      vars: [ FOO ]
    vars:
      BAR:
        sh: echo '{{.FOO}}' && false
    cmds:
      - echo Got {{.FOO}} {{.BAR }}
```

**Comment by hans-d** [comment]
related to the requires issue: #1950

**Comment by andreynering** [comment]
Hey @hans-d,

I tried your example `Taskfile.yml`, but it works fine for both tasks. I'm I missing anything?

```bash
➜  task git:(main) ✗ task test-ok
task: Task "test-ok" cancelled because it is missing required variables: FOO
➜  task git:(main) ✗ task test-fail
task: Task "test-fail" cancelled because it is missing required variables: FOO
```

**Comment by vmaerten** [comment]
Hello !

@andreynering  It has been fixed in #1962, but it's not released yet
@hans-d It'll be available in the next release

**Comment by andreynering** [comment]
@vmaerten I didn't see that PR. Thanks!
